### ISSUE TITLE
Postman Flows - No Undo - or it does not work?

### ISSUE THREAD

**Issue by Romasato** [a85/POSTMan-Chrome-Extension#13135]
### Is there an existing issue for this?

- [X] I have searched the tracker for existing similar issues and I know that duplicates will be closed

### Describe the Issue

In Flows editor, actioning "Undo" does not do anything and does not restore deleted blocks (requests) etc.

### Steps To Reproduce

1. Create new flow.
2. Add a request.
3. Delete request.
4. Trigger "Undo" - either via menu or keyboard shortcut.
5. Nothing happens - deleted block is not restored.

### Operating System

macOS

### Postman Version

11.11.1

### Postman Platform

Postman App

### User Account Type

Signed In User

**Comment by Romasato** [comment]
Is there anyone at all looking at these?

**Comment by rabbah** [comment]
@Romasato sorry for the delay in getting back to you on this. I could not reproduce this failure 
Undo should work in other words. Can you try with the latest app and post a screen recording if possible in case there's something more subtle going on?

**Comment by Romasato** [comment]
Good day @rabbah,
Thank you for following up - I have tested in latest Postman version v11.17.1 and the Undo still does not do anything in Flows editor.

Attaching the video demonstrating the issue. In the video I've also tried pressing CMD+Z with no effect.

https://github.com/user-attachments/assets/48552792-15b2-4aff-bfde-fb6eb73bedf2

P.S. There was also another separate issue with Flows what editor was showing was not reflecting the actual flow logic and it resulted in unwanted changes on our backend service - but I have no more details at the moment to raise a separate ticket as it is not easy to replicate at all.

**Comment by rabbah** [comment]
Thanks for the recording. Can you confirm if the arrows in the flows toolbar at the bottom of the canvas work for undo/redo?
<img width="503" alt="Screenshot 2024-10-18 at 8 24 15 AM" src="https://github.com/user-attachments/assets/3a516653-fc78-4d9d-a440-66a602640af8">

**Comment by rabbah** [comment]
for the second issue - is it possible there were multiple postman app.browser tabs opens to the same flow? i'm wondering if there was maybe an issue sync'ing changes.

**Comment by Romasato** [comment]
> Thanks for the recording. Can you confirm if the arrows in the flows toolbar at the bottom of the canvas work for undo/redo? 
> <img alt="Screenshot 2024-10-18 at 8 24 15 AM" width="503" src="https://private-user-images.githubusercontent.com/4959922/377865134-3a516653-fc78-4d9d-a440-66a602640af8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjkyNTgxNDcsIm5iZiI6MTcyOTI1Nzg0NywicGF0aCI6Ii80OTU5OTIyLzM3Nzg2NTEzNC0zYTUxNjY1My1mYzc4LTRkOWQtYTQ0MC02NmE2MDI2NDBhZjgucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTAxOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEwMThUMTMyNDA3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NGYzZTM0YWIzNTcwODRlMjc5ZjBjNTEzYzlmYmZiNjM5ZjMzN2U3NWFkMTljYTM3ODVkZGYzY2IyODYwZjc0MiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.Jjia4UXA6r4InlhTB3wV-H6hNXywEU0RaXf5kJ0fKQE">

Yes, the Undo in Flows toolbar does work - albeit for some reason requires 2 clicks to undo a single "delete" change.

Looks like neither app menus nor keyboard shortcuts work in terms of Undo or Delete. Only icon buttons within the Flows viewport do.


https://github.com/user-attachments/assets/cf0e2c5a-ffbf-4340-a099-400d5e797fd4

RE: The "sync" issue: it could have been 2 tabs for the same workflow open, though not sure how to get into that state. I may have also Duplicated a flow from another one - could be the references didn't get updated or something along those lines?

**Comment by rabbah** [comment]
Thanks for confirming - we're addressing the menu/keyboard shortcut bug.

**Comment by Romasato** [comment]
Awesome!

Meanwhile I'll try and replicate the sync issue, will let you know if I manage to.

**Comment by katie-roberts-alight** [comment]
I get sync issues even without having the flow open in multiple instances, I think because I'm hitting undo and redo a lot and quickly in succession trying to get it to work.

**Comment by rabbah** [comment]
I'm going to close this issue. Improving keyboard shortcuts is outstanding and covered by #12219. The syncing issue is a bug we have addressed in previous update, and have an upcoming update that will further harden simulteaneous edits.

**Comment by Romasato** [comment]
Thanks @rabbah - I haven't tried much to replicate the sync issue, but I had no real chance to use Flows lately. 
Can close - thanks for looking into this.
